core/trino-main/src/main/java/io/trino/operator/PagesIndex.java

```diff
@@ -133,19 +133,25 @@ private PagesIndex(
     {
         public static final TypeOperators TYPE_OPERATORS = new TypeOperators();
         private static final OrderingCompiler ORDERING_COMPILER = new OrderingCompiler(TYPE_OPERATORS);
-        private static final JoinCompiler JOIN_COMPILER = new JoinCompiler(TYPE_OPERATORS);
+        private final JoinCompiler joinCompiler;
```

Comment:
Will this not make JoinCompiler to be created per session, thus breaking the caching across sessions?

Reply:
TBH I don't know. I did this to make testing possible. cc @sopel39
But that would somehow explain why the benchmarks suck so much

Reply:
> But that would somehow explain why the benchmarks suck so much

Why would that impact benchmarks? You mean micro benchmarks? Could we extract `TestingFactory` as static final there?

Reply:
This is a misunderstanding, this has nothing to do with benchmarks. We discussed it today

Reply:
Just to confirm, this is test code that isn't used in actual query execution ?

Reply:
Correct. I don't support putting the test code in main, but it was there already and we decided with @sopel39 to do it like that